p2p/peermanager.go

```diff
@@ -792,6 +792,11 @@ func (m *PeerManager) Subscribe() *PeerUpdates {
 	// compounding. Limiting it to 1 means that the subscribers are still
 	// reasonably in sync. However, this should probably be benchmarked.
 	peerUpdates := NewPeerUpdates(make(chan PeerUpdate, 1))
+	m.Register(peerUpdates)
+	return peerUpdates
+}
+
+func (m *PeerManager) Register(peerUpdates *PeerUpdates) {
```

Comment:
Why was this change necessary? In any case, can we add a godoc for the new method?

Reply:
I added some godoc. 

There was no other way, to inject a PeerUpdates object into the manager (and therefore have access to the update channels) from tests. I added this to keep the tests a bit more stable, and this seems like a reasonable interface (without knowing more) but maybe I've missed something.